components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -175,13 +180,36 @@ open class GleanInternalAPI {
 
         // Build and send the ping on an async thread. Since the pingMaker.collect() function
         // returns null if there is nothing to send we can use this to avoid sending an empty ping
-        pingMaker.collect(store)?.let { pingContent ->
+        pingMaker.collect(store)?.also { pingContent ->
             val uuid = UUID.randomUUID()
             val path = makePath(docType, uuid)
-            // Asynchronously perform the HTTP upload off the main thread.
-            GlobalScope.launch(Dispatchers.IO) {
-                httpPingUploader.upload(path = path, data = pingContent)
-            }
+
+            // This holds input data for the WorkManager to access when it runs the worker task
+            // It's *important* to note Data has a hard size limit of Data.MAX_DATA_BYTES which
+            // has a constant value of 10240 according to the documentation.
```

Comment:
This seems awfully small.  What's the solution -- serialize the ping content to disk and use it right before sending?

(I vaguely recall this already being discussed, so sorry if I'm retreading ground).

Reply:
Oh geeze -- this discussion is right up there.  Ignore me...  We're all on the same page :)

Reply:
I'm working on spreading out the architecture to a ping storage engine to separate the serialization and the upload worker.  The hard part of this is testing WorkManager class, it's a bit new and the test helper library is still in beta and lacking documentation that I can find.  The best I've been able to do so far is to try and copy a minimal example embedded in the example for the WorkManager provided by the Android devs.